packages/block-library/src/query/view.js

```diff
@@ -26,9 +31,13 @@ store( 'core/query', {
 			const queryRef = ref.closest(
 				'.wp-block-query[data-wp-router-region]'
 			);
-			const isDisabled = queryRef?.dataset.wpNavigationDisabled;
+			const { clientNavigation = true } = getConfig( 'core/router' );
```

Comment:
> maybe this logic should be moved to the Router's navigate?

Yes, this needs to be done in the router.

Reply:
There will be some cases where the block needs to know if it's enabled or not, like in an instant search, but the rest of the time it should be automatic.

Reply:
> I prefer something like `clientNavigationDisabled` so we don't have to set a default value when the config is consumed.

I'm not opposed to `clientNavigationDisabled`, but can't this be true by default?

```js
const { clientNavigation } = getConfig();
if (clientNavigation !== false) {
  // navigate...
}
```

Reply:
It can―it's just that, as a boolean value, it can be confusing why the following code doesn't work:

```js
const { clientNavigation } = getConfig();
if ( clientNavigation ) {
  // Oops! It doesn't navigate when undefined, but it should!
}
```

Anyway, as long as we clarify that an `undefined` value should be treated as `true`, I'm OK with `clientNavigation`.

Reply:
I also prefer a shorter name, BTW. I'm only worried about potential confusion.

Reply:
Changed to `clientNavigationDisabled` in https://github.com/WordPress/gutenberg/pull/58749/commits/7f60ebee2da15144cfd6ce38efdd65a931c5ade5!